src/v/kafka/server/handlers/offset_fetch.cc

```diff
@@ -48,7 +48,63 @@ offset_fetch_handler::handle(request_context ctx, ss::smp_service_group) {
     request.decode(ctx.reader(), ctx.header().version);
     klog.trace("Handling request {}", request);
 
+    if (!ctx.authorized(acl_operation::describe, request.data.group_id)) {
+        co_return co_await ctx.respond(
+          offset_fetch_response(error_code::group_authorization_failed));
+    }
+
+    /*
+     * request is for all group offsets
+     */
+    if (!request.data.topics) {
+        auto resp = co_await ctx.groups().offset_fetch(std::move(request));
+        if (resp.data.error_code != error_code::none) {
+            co_return co_await ctx.respond(std::move(resp));
+        }
+
+        // remove unauthorized topics from response
+        auto unauthorized = std::partition(
+          resp.data.topics.begin(),
+          resp.data.topics.end(),
+          [&ctx](const offset_fetch_response_topic& topic) {
+              return ctx.authorized(acl_operation::describe, topic.name);
+          });
+        resp.data.topics.erase(unauthorized, resp.data.topics.end());
```

Comment:
if topics are an array this might realloc, you can simply `pop_back()`

Reply:
yeh I don't see any evidence that erase will realloc, especially given the property that iterators to previous elements are not invalidated, and that listed complexity is in terms of the number of erased elements.